### ISSUE TITLE
The google file app removed shortcut to nearby sharing

### ISSUE THREAD

**Issue by ICHx** [grishka/NearDrop#153]
The README would need to be updated accordingly.

**Comment by etekweb** [comment]
+1 for this - I will note that adding a shortcut to the `com.google.android.gms/.nearby.sharing.ReceiveSurfaceActivity` activity mentioned in the README still works (at least for me).

**Comment by fiorins** [comment]
Yes, google file removed nearby share tab so now it's impossible to do it

**Comment by eliasacab** [comment]
> +1 for this - I will note that adding a shortcut to the `com.google.android.gms/.nearby.sharing.ReceiveSurfaceActivity` activity mentioned in the README still works (at least for me).

Great solution, thanks!

**Comment by LeSchurke** [comment]
Which shortcut app did you use?

**Comment by eliasacab** [comment]
> Which shortcut app did you use?

Root Activity Launcher

**Comment by LeSchurke** [comment]
> > Which shortcut app did you use?
> 
> Root Activity Launcher

Than this? ![Screenshot_20240502-004105.png](https://github.com/grishka/NearDrop/assets/79398820/6ee07a94-f770-4f65-90b6-c8411976571f)

**Comment by etekweb** [comment]
> Which shortcut app did you use? 

I used this one: [Activity Launcher](https://play.google.com/store/apps/details?id=de.szalkowski.activitylauncher) 

Just search the activity name, then you can tap the item in the list to test it, or click the menu icon to save a home screen shortcut.

**Comment by etekweb** [comment]
> > > Which shortcut app did you use?
> > 
> > 
> > I used this one: [Activity Launcher](https://play.google.com/store/apps/details?id=de.szalkowski.activitylauncher)
> > Just search the activity name, then you can tap the item in the list to test it, or click the menu icon to save a home screen shortcut.
> 
> I didn't find it

The search term I used that worked was `com.google.android.gms.nearby.sharing.ReceiveSurfaceActivity`

**Comment by andrew1-1** [comment]
> > > > Which shortcut app did you use?
> > > 
> > > 
> > > I used this one: [Activity Launcher](https://play.google.com/store/apps/details?id=de.szalkowski.activitylauncher)
> > > Just search the activity name, then you can tap the item in the list to test it, or click the menu icon to save a home screen shortcut.
> > 
> > 
> > I didn't find it
> 
> The search term I used that worked was `com.google.android.gms.nearby.sharing.ReceiveSurfaceActivity`
> 
> It doesn't go through Files app, it goes through Google Play Services.

Thank you for the tip! In my case however I didn't find the exact the term you used, what I found instead was `com.google.android.gms.nearby.sharing.receive.ReceiveActivity` and it did work for me and I was this way able to have my phone get detected by NearDrop

**Comment by 16bitman-MSDOS** [comment]
Hi, looks like after Jan 2025, they removed this service for the pixel. I’d like to confirm with others here if it is the same case?

My activity launcher shortcut shows an error now and cannot find `com.google.android.gms.nearby.sharing.receive.ReceiveActivity`
What I did notice though was that in files app, the phone can detect that it is receiving a file through quick share (just like before with nearby, before they removed it) but it get’s stuck at “Connecting” in Mac.
Nothing else that I could find in activity launcher to bring nearby share function back.
